components/Sort.tsx

```diff
@@ -76,6 +75,13 @@ export default function Sort() {
       return SortTypes.BestMatch;
     }
   };
+
+  const handleClick = () => {
+    const elem = document.activeElement as HTMLElement;
```

Comment:
It seems like clicking the `<li>` tag makes the `activeElement` so the dropdown does not close. Can you find a way to fix that?

Reply:
You are right. I will look into the solution to this again.

Reply:
I can fix this with useState but the old functioning has broken. Please do not merge. I'll look at this again.

Reply:
Sure.

Reply:
I understand the problem better now. The part I solved is that after clicking on the dropdown menu, the dropdown menu continues to appear even after leaving the item. Now I click on the dropdown menu and the page content changes. If the mouse is over the dropdown menu, this item continues to appear. As far as I researched, attempts were made to solve it with onclick and state. I also tried various solutions. For example, onmouseover was one of them. When I tried this solution, I noticed that there was a problem with the animation of this element when I moved to the dropdown with the mouse, changed the page content, and then hovered over the dropdown again. That's why I shelved this solution. If you want, I can open another issue about this problem and someone who knows better can solve it.

Reply:
What do you think of this solution?
https://stackblitz.com/edit/nextjs-daisy-ui-v3pfmf?file=components%2FDropdown.js

Reply:
Yes, this seems to be the most optimal and only solution to this problem.  Because this problem is caused by daisy ui, but daisy ui does not offer a solution for it.  The current code structure is not suitable for solving this problem in any other way.

Reply:
The bug you mentioned is caused by the fact that the main container also has a dropdown-hover class. To solve this situation, I gave the button onmouseover, but I saw that there was a problem with the animation of the dropdown.

![image](https://github.com/max-programming/hacktoberfest-projects/assets/80166639/b16287ed-8e9e-4341-baa3-7e4a3269470d)

Reply:
It's okay anyways there is no exit animation when we click outside. Implement it and if it feels annoying, we'll see